agenta-cli/agenta/sdk/decorators/llm_entrypoint.py

```diff
@@ -73,31 +102,69 @@ async def chain_of_prompts_llm(prompt: str):
     ```
     """
 
-    def __init__(self, func: Callable[..., Any]):
+    routes = list()
+
+    def __init__(self, func: Callable[..., Any], route_path=""):
         endpoint_name = "generate"
+        playground_path = "/playground"
+        run_path = "/run"
         func_signature = inspect.signature(func)
-        config_params = agenta.config.all()
+        config_params = ag.config.all()
         ingestible_files = self.extract_ingestible_files(func_signature)
 
+        ### --- Playground / Drafts  --- #
         @debug()
         @functools.wraps(func)
         async def wrapper(*args, **kwargs) -> Any:
             func_params, api_config_params = self.split_kwargs(kwargs, config_params)
             self.ingest_files(func_params, ingestible_files)
-            agenta.config.set(**api_config_params)
+            ag.config.set(**api_config_params)
 
             # Set the configuration and environment of the LLM app parent span at run-time
-            agenta.tracing.update_baggage(
+            ag.tracing.update_baggage(
                 {"config": config_params, "environment": "playground"}
             )
 
             # Exceptions are all handled inside self.execute_function()
-            llm_result = await self.execute_function(
+            entrypoint_result = await self.execute_function(
                 func, *args, params=func_params, config_params=config_params
             )
 
-            return llm_result
+            return entrypoint_result
 
+        self.update_function_signature(
+            wrapper, func_signature, config_params, ingestible_files
+        )
+
+        # TODO:
+        # - The whole ag.Config is now required for individual stages.
+        #   Once ag.Config goes from Singleton to Instance, and
+        #   depending on how ag.Config is implemented,
+        #   we need to filter the part of ag.Config that matters to each route.
+
+        if route_path == "":
+            route = f"/{endpoint_name}"
+            app.post(route, response_model=BaseResponse)(wrapper)
+            entrypoint.routes.append(
```

Comment:
Because `routes` is a class variable and not an instance variable.
Every time `@entrypoin` and/or `@route()` are used, the `entrypoint.routes` class variable is updated.